### PR TITLE
A core file generates when resizing the volume during R/W with vdbench

### PR DESCRIPTION
**Pull request by inspur001**
rbd: assert happens when resizing the image during testing

when continous R/W being in progress, resizing the image will chage the size of the object-map, but the correspoding info in the memory does not update, so memory-access errors will occur and the program will go down 

Signed-off-by: yanghongjie <yanghongjie@inspur.com>

### CODE REVIEW

File: src/librbd/ObjectMap.h [hunk 1]
```diff
@@ -69,6 +69,9 @@ class ObjectMap {
                   const boost::optional<uint8_t> &current_state,
                   const ZTracer::Trace &parent_trace, bool ignore_enoent,
                   T *callback_object) {
+  	if (start_object_no >= m_object_map.size()){
```
Comment (tchaikov) [inline src/librbd/ObjectMap.h]: bad indent.

File: src/librbd/ObjectMap.cc [hunk 2]
```diff
@@ -81,7 +81,9 @@ template <typename I>
 bool ObjectMap<I>::object_may_exist(uint64_t object_no) const
 {
   ceph_assert(m_image_ctx.image_lock.is_locked());
-
+  if (object_no >= m_object_map.size()){
+  	return true;
```
Comment (tchaikov) [inline src/librbd/ObjectMap.cc]: bad indent.

File: src/librbd/ObjectMap.h (hunk 1 above)
Comment (inspur001) [reply src/librbd/ObjectMap.h]: done

File: src/librbd/ObjectMap.cc (hunk 2 above)
Comment (inspur001) [reply src/librbd/ObjectMap.cc]: done

**Comment by dillaman** [comment]
@inspur001 Was this issue fixed by PR #28295? The original issue was that IO was resumed prior to growing the object map, but the PR addressed that by keeping IO paused until after.

**Comment by dillaman** [comment]
I believe this issue has been resolved so I am closing this PR. If it's not resolved, feel free to re-open.

**Pull request closed without merging**